modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -709,8 +709,16 @@ class NaryEltwiseLayerImpl CV_FINAL : public NaryEltwiseLayer
             }
             case OPERATION::MEAN:
             {
-                auto mean = [](const T &a, const T &b) { return (a + b) / T{2}; };
-                nary_forward<T>(mean, T{1} / ninputs, std::forward<Args>(args)...);
+                if (ninputs == 2){
```

Comment:
Why do we need to handle ninputs=2 and ninputs=3 cases? Can we just use nary_forward for any number of inputs?